lib/tasks/helpers/importmap_helper.rb

```diff
@@ -9,43 +9,66 @@ class Packager < Importmap::Packager
     self.endpoint = URI("https://api.jspm.io/generate")
 
     # Copied from https://github.com/rails/importmap-rails/pull/237
-    def verify(package, url)
+    def verify(package, url, verbose: false)
       ensure_vendor_directory_exists
 
-      return unless vendored_package_path(package).file?
-      verify_vendored_package(package, url)
+      unless vendored_package_path(package).file?
+        raise ImportmapHelper::VerifyError, "Pinned #{package}#{extract_package_version_from(url)} does not exist in vendor/javascript"
+      end
+      verify_vendored_package(package, url, verbose:)
     end
 
-    def verify_vendored_package(package, url)
+    def verify_vendored_package(package, url, verbose: false)
       vendored_body = vendored_package_path(package).read.strip
-      remote_body = load_package_file(package, url).strip
+      vendored_body = vendored_body.lines[2..].join if vendored_body.start_with?("//") # remove the importmap-rails comment
+      remote_body = load_package_file(url).strip
 
       return true if vendored_body == remote_body
 
-      raise ImportmapHelper::VerifyError, "Vendored #{package}#{extract_package_version_from(url)} does not match remote #{url}"
+      verbose_error = verbose ? verbose_diff(remote_body, vendored_body) : " (run with VERBOSE=true for diff)"
+      raise ImportmapHelper::VerifyError, "Vendored #{package}#{extract_package_version_from(url)} does not match remote #{url}#{verbose_error}"
     end
 
-    def load_package_file(package, url)
+    def load_package_file(url)
       response = Net::HTTP.get_response(URI(url))
 
       if response.code == "200"
-        format_vendored_package(package, url, response.body)
+        format_vendored_package(response.body)
       else
         handle_failure_response(response)
       end
     end
 
-    def format_vendored_package(package, url, source)
-      formatted = +""
-      if Gem::Version.new(Importmap::VERSION) > Gem::Version.new("2.0.1")
-        formatted.concat "// #{package}#{extract_package_version_from(url)} downloaded from #{url}\n\n"
+    def format_vendored_package(source)
+      remove_sourcemap_comment_from(source).force_encoding("UTF-8")
+    end
+
+    def save_vendored_package(package, url, source)
+      File.open(vendored_package_path(package), "w+") do |vendored_package|
+        vendored_package.write "// #{package}#{extract_package_version_from(url)} downloaded from #{url}\n\n"
+
+        vendored_package.write remove_sourcemap_comment_from(source).force_encoding("UTF-8")
       end
-      formatted.concat remove_sourcemap_comment_from(source).force_encoding("UTF-8")
-      formatted
     end
 
-    def save_vendored_package(package, _url, source)
-      File.write(vendored_package_path(package), source)
+    # The built in version matcher doesn't handle pre-release versions.
+    # This causes mismatches and downloads the wrong version on refetch.
+    def extract_package_version_from(url)
```

Comment:
This method breaks rubocop. But is there any reason why it is needed? Which pre-release is used? Is this something we can send to upstream?

Reply:
`@rails/ujs` causes the version mismatch problem. I would like to push this upstream.

Reply:
Btw, this is the server.org regex.

Reply:
But it is not pre-release, right? It is just x.y.z-a kind of versioning.

Reply:
The -a is considered a pre-release and is ignored by the previous regex, causing the url and version stored in the comment to be wrong.

Reply:
I mean we're not using pre-release of any library in production. It is just different versioning schema.

Reply:
Yes, not "prerelease" in production, but rails releases ujs using the prerelease formatting for semver which is not exactly supported in importmap-rails (it downloads the x.y.z-4 version on pin but reports that it downloaded x.y.z and when redownloading it grabs a slightly different version)

Reply:
I dropped this line for now and I'll try to push it upstream.